agent/cache/cache.go

```diff
@@ -279,6 +279,7 @@ func (c *Cache) getWithIndex(t string, r Request, minIndex uint64) (interface{},
 
 RETRY_GET:
 	// Get the type that we're fetching
+	// TODO: should this lookup of the type be before RETRY_GET?
 	c.typesLock.RLock()
 	tEntry, ok := c.types[t]
 	c.typesLock.RUnlock()
```

Comment:
Thoughts on this TODO?

I believe we register the types once at startup and they never change. They are effectively static, and I believe that a cache entry can never change types. So could we move this lookup out of the "loop" and avoid getting the lock?

Reply:
This seems reasonable to me, don't see any other places we call `RegisterType` besides testing functions.